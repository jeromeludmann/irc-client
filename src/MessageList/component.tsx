import React from "react";
import { ChannelScope } from "@app/types";

type Props = ChannelScope & {
  messages: string[];
};

export class MessageList extends React.PureComponent<Props> {
  public render() {
    return (
      <ul>
        {this.props.messages.map(message => {
          return <li key={message}>{message}</li>;
        })}
      </ul>
    );
  }
}
