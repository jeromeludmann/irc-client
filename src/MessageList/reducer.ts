import { ChannelScope } from "@app/types";

export type MessageListState = string[];

interface MessageListAction {
  type: "COMMAND_SENT";
  payload: ChannelScope & { value: string };
}

const messagesInitialState: MessageListState = [];

export const messageListReducer = (
  state = messagesInitialState,
  { type, payload }: MessageListAction,
): MessageListState => {
  switch (type) {
    case "COMMAND_SENT":
      return [...state, payload.value];
    default:
      return state;
  }
};
