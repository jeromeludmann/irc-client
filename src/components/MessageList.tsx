import React from "react";

interface Props {
  messages: string[];
}

export default ({ messages }: Props) => (
  // TODO do not use index
  <ul
    style={{
      fontFamily: "Menlo, Monaco, Courier",
      fontSize: "12px",
      listStyle: "none",
      padding: "0",
    }}
  >
    {messages.map((message, i) => <li key={i}>{message}</li>)}
  </ul>
);
