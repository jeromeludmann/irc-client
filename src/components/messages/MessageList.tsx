import React from "react";
import { MessageList, MessageListItem } from "@app/components/messages/style";

interface Props {
  messages: string[];
}

export default ({ messages }: Props) => (
  // TODO
  // do NOT use index as a key
  // instead generate a timestamp on each message
  <MessageList>
    {messages.map((message, i) => (
      <MessageListItem key={i}>{message}</MessageListItem>
    ))}
  </MessageList>
);
