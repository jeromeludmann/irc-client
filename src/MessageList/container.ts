import { connect } from "react-redux";
import { RootState } from "@app/reducers/rootReducer";
import { MessageList as Component } from "@app/MessageList/component";
import { ChannelScope } from "@app/types";
import { getMessages } from "@app/MessageList/selectors";

type OwnProps = ChannelScope & {};

interface StateProps {
  messages: string[];
}

const mapStateToProps = (state: RootState, ownProps: OwnProps): StateProps => ({
  messages: getMessages(state, {
    server: ownProps.server,
    channel: ownProps.channel,
  }),
});

export const MessageList = connect<StateProps, void, OwnProps>(mapStateToProps)(
  Component,
);
