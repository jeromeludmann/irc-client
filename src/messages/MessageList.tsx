import React from "react";
import { connect } from "react-redux";
import { RootState } from "@app/state";
import { getMessages } from "@app/state/channel/messages";

interface OwnProps {
  server: string;
  channel: string;
}

interface StateProps {
  messages: string[];
}

const MessageList = ({ messages }: OwnProps & StateProps) => (
  <ul style={{ listStyle: "none", padding: "5px 10px" }}>
    {messages.map(message => <li key={message}>{message}</li>)}
  </ul>
);

const mapStateToProps = (
  state: RootState,
  { server, channel }: OwnProps,
): StateProps => ({ messages: getMessages(state, server, channel) });

export default connect<StateProps, void, OwnProps>(mapStateToProps)(
  MessageList,
);
