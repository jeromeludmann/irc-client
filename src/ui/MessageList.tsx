import React from "react";
import { connect } from "react-redux";
import { RootState } from "@app/state";
import { getMessages } from "@app/state/channel/messages";

interface OwnProps {
  server: string;
  channel: string;
}

interface StateProps {
  messages: Message[];
}

interface Message {
  timestamp: number;
  value: string;
}

const MessageList = ({ messages }: OwnProps & StateProps) => {
  return (
    <ul style={{ listStyle: "none", padding: "5px 10px" }}>
      {messages.map(message => (
        <li key={message.timestamp}>
          {message.timestamp} | {message.value}
        </li>
      ))}
    </ul>
  );
};

const mapStateToProps = (
  state: RootState,
  { server, channel }: OwnProps,
): StateProps => ({ messages: getMessages(state, server, channel) });

export default connect<StateProps, void, OwnProps>(mapStateToProps)(
  MessageList,
);
