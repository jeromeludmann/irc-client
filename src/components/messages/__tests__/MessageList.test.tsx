import React from "react";
import { shallow } from "enzyme";
import MessageList from "@app/components/messages/MessageList";

describe("MessageList component", () => {
  const messages = ["hello", "world"];

  it("should render correctly", () => {
    expect(shallow(<MessageList messages={messages} />)).toMatchSnapshot();
  });
});
