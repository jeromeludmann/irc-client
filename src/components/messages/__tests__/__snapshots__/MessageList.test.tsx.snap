// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    messages={
      Array [
        "hello",
        "world",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.li>
          hello
        </styled.li>,
        <styled.li>
          world
        </styled.li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": "hello",
        },
        "ref": null,
        "rendered": "hello",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": "world",
        },
        "ref": null,
        "rendered": "world",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.li>
            hello
          </styled.li>,
          <styled.li>
            world
          </styled.li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": "hello",
          },
          "ref": null,
          "rendered": "hello",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": "world",
          },
          "ref": null,
          "rendered": "world",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
